TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/CaptureProxy.java

```diff
@@ -191,19 +195,20 @@ static Parameters parseArgs(String[] args) {
 
     @SneakyThrows
     protected static Settings getSettings(@NonNull String configFile) {
-        var builder = Settings.builder();
-        try (var lines = Files.lines(Paths.get(configFile))) {
-            lines.map(
-                line -> Optional.of(line.indexOf('#')).filter(i -> i >= 0).map(i -> line.substring(0, i)).orElse(line)
-            ).filter(line -> line.startsWith(HTTPS_CONFIG_PREFIX) && line.contains(":")).forEach(line -> {
-                var parts = line.split(": *", 2);
-                builder.put(parts[0], parts[1]);
-            });
-        }
-        builder.put(SSLConfigConstants.SECURITY_SSL_TRANSPORT_ENABLED, false);
+        var objectMapper = new ObjectMapper(new YAMLFactory());
+        var configMap = objectMapper.readValue(new File(configFile), Map.class);
+
         var configParentDirStr = Paths.get(configFile).toAbsolutePath().getParent();
-        builder.put("path.home", configParentDirStr);
-        return builder.build();
+        var httpsSettings =
+            objectMapper.convertValue(configMap, new TypeReference<Map<String, Object>>(){})
+                .entrySet().stream()
+                .filter(kvp -> kvp.getKey().startsWith(HTTPS_CONFIG_PREFIX))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        return Settings.builder().loadFromMap(httpsSettings)
+            .put(SSLConfigConstants.SECURITY_SSL_TRANSPORT_ENABLED, false)
```

Comment:
This looks like we're always setting TLS to false -- is that true? Why?

Reply:
Because that's for the 9300 port setup stuff, which the proxy doesn't use.  I'll add a comment to explain.